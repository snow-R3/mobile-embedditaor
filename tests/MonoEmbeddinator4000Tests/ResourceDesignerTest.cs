using System;
using System.IO;
using ApprovalTests;
using ApprovalTests.Reporters;
using NUnit.Framework;

namespace MonoEmbeddinator4000.Tests
{
    /// <summary>
    /// These are a set of integration/approval tests validating that we are getting expected C# code from ResourceDesignerGenerator
    /// </summary>
    [TestFixture, UseReporter(typeof(NUnitReporter))]
    public class ResourceDesignerTest
    {
        ResourceDesignerGenerator generator;

        [SetUp]
        public void SetUp()
        {
            generator = new ResourceDesignerGenerator
            {
                MonoDroidPath = Samples.MonoDroidPath,
                OutputDirectory = Environment.CurrentDirectory,
                PackageName = "com.mono.embeddinator",
            };
        }

        [TearDown]
        public void TearDown()
        {
            //Temp file
            File.Delete(generator.MainAssembly);
        }

        void LoadAndGenerate(string resourceFile)
        {
            var assembly = Samples.LoadFile(resourceFile);
            generator.Assemblies = new[] { assembly };
            generator.MainAssembly = assembly.Location;
            generator.Generate();
        }

        [Test]
        public void String()
        {
            LoadAndGenerate("String");

            string source = generator.ToSource();
            Approvals.Verify(source);
        }

        [Test]
        public void Full()
        {
            LoadAndGenerate("Full");

            string source = generator.ToSource();
            Approvals.Verify(source);
        }
    }
}
