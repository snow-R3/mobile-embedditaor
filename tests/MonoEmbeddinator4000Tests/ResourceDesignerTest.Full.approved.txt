//------------------------------------------------------------------------------
// <auto-generated>
//     This code was generated by a tool.
//     Runtime Version:4.0.30319.42000
//
//     Changes to this file may cause incorrect behavior and will be lost if
//     the code is regenerated.
// </auto-generated>
//------------------------------------------------------------------------------

[assembly: Android.Runtime.ResourceDesignerAttribute("__embeddinator__.Resource", IsApplication=true)]

namespace @__embeddinator__ {
    using System;
    using Android.Runtime;
    
    
    [System.CodeDom.Compiler.GeneratedCodeAttribute("Xamarin.Android.Build.Tasks", "1.0.0.0")]
    public class Resource {
        
        private static Int32 ReadFieldInt(IntPtr R, String fieldName) {
            IntPtr fieldId = JNIEnv.GetStaticFieldID(R, fieldName, "I");
            return JNIEnv.GetStaticIntField(R, fieldId);
        }
        
        private static Int32[] ReadFieldArray(IntPtr R, String fieldName) {
            IntPtr fieldId = JNIEnv.GetStaticFieldID(R, fieldName, "[I");
            IntPtr value = JNIEnv.GetStaticObjectField(R, fieldId);
            return JNIEnv.GetArray<Int32>(value);
        }
        
        public static void UpdateIdValues() {
            IntPtr R;
            R = JNIEnv.FindClass("com.mono.embeddinator.R$attr");
            managedandroid.Resource.Attribute.hello = ReadFieldInt(R, "hello");
            R = JNIEnv.FindClass("com.mono.embeddinator.R$id");
            managedandroid.Resource.Id.text = ReadFieldInt(R, "text");
            R = JNIEnv.FindClass("com.mono.embeddinator.R$layout");
            managedandroid.Resource.Layout.customView = ReadFieldInt(R, "customview");
            managedandroid.Resource.Layout.hello = ReadFieldInt(R, "hello");
            R = JNIEnv.FindClass("com.mono.embeddinator.R$string");
            managedandroid.Resource.String.hello = ReadFieldInt(R, "hello");
            managedandroid.Resource.String.library_name = ReadFieldInt(R, "library_name");
            R = JNIEnv.FindClass("com.mono.embeddinator.R$styleable");
            managedandroid.Resource.Styleable.Theme = ReadFieldArray(R, "Theme");
            managedandroid.Resource.Styleable.Theme_hello = ReadFieldInt(R, "Theme_hello");
        }
    }
}
